### ISSUE TITLE
compute sound-class and word length statistics

### ISSUE THREAD

**Issue by LinguList** [glottobank/lexibank-data#46]
given that this is the ideal playground for lingpy and similar approaches, we should try and see what we actually can do with this.

* lingpy offers sound class models in different flavors, but even a pie-chart of dolgopolsky-classes may give an interesting picture on the importance of sounds in a language, this can be implemented in a straightforward way, and the meaning is, like: 20% of all sound segments (tokens) are velar, 40% are vocalic, etc.
* even cooler are metrics on word length. In lingpy, we can easily compute the number of consonants, the number of vowels per word, the number of syllables, and the number of sounds

```python
>>> from lingpy.sequence.sound_classes import syllabify, clean_string, tokens2class
>>> import lingpy as lp
>>> syllable_length = len(syllabify('m a t a t a', output='nested'))
>>> consonant_length = len([x for x in tokens2class('m a t a t a'.split(' '), 'dolgo') if x != 'V'])
>>> print('etcetera...')
```

Just FYI, I tested the syllable and consonant length and it's impact on cognate detection, and I found that the Dolgopolsky method performs worst on the dataset where we have more than 3 consonants on average per word. So there seems to be something to this...

**Comment by SimonGreenhill** [comment]
Cool, that is interesting.  Further down the track it may be fun to check *computed* phoneme inventories from the wordlists to the ones in Phoible...

**Comment by LinguList** [comment]
And the cool, very cool addition is that we may also include phonotactic information (how many intial consonants, maximal number of consonant clusters, etc.), and we might get a lot of information from that, which might both help us improving cognate detection (think of thresholds based on phonetic diversity, for example), but also get new insights into some characteristics of evolution. We just need to think of interesting measure we might want to collect, and think of ways how to implement them.

Syllable length, for example, is rarely counted in datasets, as most linguists agnostically assume that one could not comute syllables. But the predictor I have, using syllabification based on prosody, works fairly well in probably more than 90% of all cases, so it's a cool thing to look into. We can even think of improving Turchin/Dolgopolsky-sound-class method by adding syllable-criterions (being more restrictive if more syllables are in a word, or something similar). Just a few thoughts...

**Comment by xrotwang** [comment]
I think we should find a good explicit place in the codebase for this kind of advanced functionality. So I wouldn't expect this to be computed regularly as part of the general workflow, but rather with some particular research question in mind. So maybe the best place for it may be LingPy plugins?

Btw.: In any case, we should aim for good lexibank-data integration in LingPy for the 3.0 relase.

**Comment by LinguList** [comment]
yes, lingpy is probably the best place. We need to write parsers for CLDF, but this was anyway on the agenda, and improving on the quality checks is also crucial for automatic cognate detection, as it would be cool to have a good integration of lexibank, concepticon, and clpa.
